app/controllers/admin/publishers_controller.rb

```diff
@@ -2,14 +2,43 @@ class Admin::PublishersController < AdminController
   def index
     @publishers = Publisher
     if params[:q].present?
-      @publishers = @publishers.where("email ILIKE :email OR name ILIKE :name",
-                      {
-                        email: "%#{params[:q]}%",
-                        name: "%#{params[:q]}%"
-                      }
-                    )
+      # Returns an ActiveRecord::Relation of publishers for pagination
+      @publishers = Publisher.where("publishers.id IN (#{sql(params[:q])})").distinct
     end
-
     @publishers = @publishers.paginate(page: params[:page])
   end
 end
+
+private
+
+# Returns an array of publisher ids that match the query
+def sql(query)
+  %{SELECT publishers.id 
```

Comment:
I'm nitpicking, but would you change to `SELECT DISTINCT publishers.id` here? The `DISTINCT` at the top runs it at the top layer.

Reply:
💯

Reply:
Actually, I think we want this to run at the top.  Assuming we don't, if there is a publisher has a channel that matches the query AND has a name/email that matches the query, I think we'd display them twice in the results.